kapre/augmentation.py

```diff
@@ -0,0 +1,85 @@
+import tensorflow as tf
+from tensorflow.keras import backend as K
+from . import backend
+from .backend import _CH_FIRST_STR, _CH_LAST_STR, _CH_DEFAULT_STR
+from tensorflow.python.ops import array_ops
+from tensorflow.keras.layers import Layer
+from tensorflow.python.keras.utils import tf_utils
+
+__all__ = [
+    'SpecAugment'
+]
+
+class SpecAugment(Layer):
+    """
+    # `SpecAugment`
+
+    ```python
+    kapre.augmentation.SpecAugment(time_param=10, freq_param=10, **kwargs)
+    ```
+
+    Add masking to input data and output it
+    Args:
+        x: Spectrogram
+        freq_param: Param of freq masking
+        time_param: Param of Time masking
+    Returns: 
+        Masked Tensor of Spectrogram
+    """
+    def __init__(
+        self, 
+        freq_param = None, 
+        time_param = None,
+        input_data_format:str='default',
+        **kwargs
+    ):
+        
+        super(SpecAugment, self).__init__(**kwargs)
+
+        backend.validate_data_format_str(input_data_format)
+
+        self.freq_param = freq_param
+        self.time_param = time_param
+        self.uses_learning_phase = True
+        self.supports_masking = True
+
+        assert (freq_param is not None) or (time_param is not None), "at least one param value should be defined"
+        
+        idt = input_data_format
+        self.input_data_format = K.image_data_format() if idt == _CH_DEFAULT_STR else idt
+
+    def call(self, x, training=None):
+        
+        if training is None:
+            training = K.learning_phase()
```

Comment:
Do you know if this is necessary? I naively assumed if `training` argument is included in `call()` method, Keras would deal with everything correctly.

Reply:
It is training flag is used in smart_cond and could be modified on Keras Model Subclassing

Reply:
Yup and I assume, from https://www.tensorflow.org/guide/keras/custom_layers_and_models#privileged_training_argument_in_the_call_method, we don't need this if statement.

Reply:
I just followed TensorFlow practices  https://github.com/tensorflow/tensorflow/blob/b36436b087bd8e8701ef51718179037cccdfc26e/tensorflow/python/keras/layers/core.py#L206 , I can remove it though

Reply:
ha, this is interesting. never mind, let's keep it as it is to be safe.

Reply:
thanks for the context :) i'll maybe change my other code like that as well.